web/src/SidebarResources.tsx

```diff
@@ -48,6 +49,72 @@ export function SidebarListSection(
   )
 }
 
+function SidebarItemsView(props: SidebarProps) {
+  return (
+    <>
+      {props.items.map((item) => (
+        <SidebarItemView
+          key={"sidebarItem-" + item.name}
+          item={item}
+          selected={props.selected === item.name}
+          pathBuilder={props.pathBuilder}
+          resourceView={props.resourceView}
+        />
+      ))}
+    </>
+  )
+}
+
+function SidebarLabelListSection(props: { label: string } & SidebarProps) {
+  if (props.items.length === 0) {
+    return null
+  }
+
+  return (
+    <>
+      <SidebarListSectionName>{props.label}</SidebarListSectionName>
+      <SidebarListSectionItems>
+        <SidebarItemsView {...props} />
+      </SidebarListSectionItems>
+    </>
+  )
+}
+
+function SidebarGroupedByLabels(props: SidebarProps) {
+  const labelsToResources: { [key: string]: SidebarItem[] } = {}
+  const unlabeled: SidebarItem[] = []
+
+  props.items.forEach((item) => {
+    if (item.labels.length) {
+      item.labels.forEach((label) => {
+        if (!labelsToResources.hasOwnProperty(label)) {
+          labelsToResources[label] = []
+        }
+
+        labelsToResources[label].push(item)
+      })
+    } else {
+      unlabeled.push(item)
+    }
+  })
+
+  const labels = Object.keys(labelsToResources)
```

Comment:
how much do we care about the order of labels? feels weird to implicitly rely on whatever the JS implementation order is.

Reply:
hmm, how do you feel about alphabetizing the labels? that way, labels show up consistently in the same order (which i would find helpful as a user and as a user, i'd be able to manipulate the alphabet system if i want my labels to display in a specific order). otherwise, in theory, the labels will appear in the order of whatever resources we iterate over first.

Reply:
i went ahead and added label sorting code! we can revisit if there's a better suggestion for sorting.